### DISCUSSION TITLE
Stereoscopic images should pop out

### DISCUSSION THREAD

**Discussion by Vega-KH** [thygate/stable-diffusion-webui-depthmap-script#412] [Ideas]
To my eyes, it seems like stereo images produced by this extension only go inward, like I am looking through a sheet of glass into a room on the other side. For a better 3D effect, the closest things in the image should protrude through the near plane.

Has anyone tried the app at owl3d.com ? (there is a free version for lower res) Owl3d seems to use very similar techniques to convert 2d images to 3d. Their fill technique is worse than "polylines sharp" and there are some glitches, but the images pop out much better. If you try it, set the pop-out effect to level 2 for a big effect, but even with it turned off there is WAY more pop out. 

I'm not sure the method for implementing this, but I think it would make nicer images.

**Comment by sonakravtsenko3** [comment]
Try increasing the Divergence (stereo effect) values. At some point in the past the extension generated stereoimages backwards - that is, they were "inwards" and you describe. I think I fixed it. Please upgrade the extension just in case - the version should be v0.4.6.

Could you attach the image here? The depthmap you are using could also be useful - do you generate it or is it custom? If generated, please specify what model. If custom, it may be inversed.

Sorry I took long to respond.

**Comment by Vega-KH** [comment]
Hello, thanks for the reply. 

I've updated to the latest commit and generated test images with both this extension and with Owl3D. I am not using any custom model, just basic res101 with boost. This time I left all of the settings at default value (divergence 2.5, magic exponent 2). 

In many respects, the images produced by this extension are superior, especially when it comes to filling the gaps. But I think you can see the popout effect is far superior on the owl3d images. The camera lens jumps out of the image. The leading edge of the piano seems to protrude. Or maybe it's just my eyes and I'm crazy. Let me know what you think.

![photo_fullsbs (0)_owl3d](https://github.com/thygate/stable-diffusion-webui-depthmap-script/assets/3310674/4f1a4069-3858-48f3-8295-7f00de8598af)
Made with Owl3D

![photo_fullsbs (0)_extension](https://github.com/thygate/stable-diffusion-webui-depthmap-script/assets/3310674/f9053b36-fa6a-4d5e-92f5-2c1f0edb0008)
Made with WebUI extension

![photo_fullsbs (1)_owl3d](https://github.com/thygate/stable-diffusion-webui-depthmap-script/assets/3310674/abae64db-d039-4953-b637-68851b3812ff)
Made with Owl3D

![photo_fullsbs (1)_extension](https://github.com/thygate/stable-diffusion-webui-depthmap-script/assets/3310674/161ffc83-668b-4750-9654-5512658dccbd)
Made with WebUI extension

**Reply by FancyJustice** [reply]
I believe Owl3D now uses DepthAnything as it's preprocessor. I'm not sure if that would make a difference. I've also tried both and get more pop out with Owl3D but it has a setting for it and I'm not sure what it does or what the equivalent is in this extension.

**Reply by salexx** [reply]
No you are not crazy)))
I see the difference too.
If you understand what the secret is, please share! )

**Reply by redman4585** [reply]
The secret is... 

The type of depth map used: **Depth Anything**.  Res101 Boost flattens foreground elements while providing detailed backgrounds. You see the difference in the shading from the guy's face to the camera in his hand? You notice the pop better in Depth Anything because the face is pushed further back. Even if you used the same 3D strength/convergence setting you could not replicate the foreground pop out effect due to the differences in the AI depth maps.

![68,0265d00317352840](https://github.com/thygate/stable-diffusion-webui-depthmap-script/assets/126682588/ab7481d3-8158-4799-9ed1-e32281cdfa60)
RGB Image

![LERES](https://github.com/thygate/stable-diffusion-webui-depthmap-script/assets/126682588/77f762ca-331f-4751-9964-a2d39cb23aef)
Res101 Boost

![DepthAnything](https://github.com/thygate/stable-diffusion-webui-depthmap-script/assets/126682588/ad618cfa-c6fb-48e4-9aed-c33ff0e5e181)
Depth Anything

Fun with convergence: From what I see in the OP's pic, the image is not correctly aligned. Only the camera and his hands should be popping out, with the guy's face in focus. You can edit this in StereoPhoto Maker by moving the slider to the right or left in Easy Adjustment.

![Before](https://github.com/thygate/stable-diffusion-webui-depthmap-script/assets/126682588/106512dc-6444-4085-b000-12717c16a758)
Original

![After](https://github.com/thygate/stable-diffusion-webui-depthmap-script/assets/126682588/c1846721-c00c-4aa6-89dc-8558bdf647ff)
Edited
